flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/shuffle/TestDataStatisticsCoordinator.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.flink.runtime.jobgraph.OperatorID;
+import org.apache.flink.runtime.operators.coordination.EventReceivingTasks;
+import org.apache.flink.runtime.operators.coordination.MockOperatorCoordinatorContext;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.runtime.typeutils.RowDataSerializer;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.VarCharType;
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestDataStatisticsCoordinator {
+  private static final String OPERATOR_NAME = "TestCoordinator";
+  private static final OperatorID TEST_OPERATOR_ID = new OperatorID(1234L, 5678L);
+  private static final int NUM_SUBTASKS = 2;
+  private TypeSerializer<DataStatistics<MapDataStatistics, Map<RowData, Long>>>
+      statisticsSerializer;
+
+  private EventReceivingTasks receivingTasks;
+  private DataStatisticsCoordinator<MapDataStatistics, Map<RowData, Long>>
+      dataStatisticsCoordinator;
+
+  @Before
+  public void before() throws Exception {
+    receivingTasks = EventReceivingTasks.createForRunningTasks();
+    statisticsSerializer =
+        MapDataStatisticsSerializer.fromKeySerializer(
+            new RowDataSerializer(RowType.of(new VarCharType())));
+
+    dataStatisticsCoordinator =
+        new DataStatisticsCoordinator<>(
+            OPERATOR_NAME,
+            new MockOperatorCoordinatorContext(TEST_OPERATOR_ID, NUM_SUBTASKS),
+            statisticsSerializer);
+  }
+
+  private void tasksReady() throws Exception {
+    dataStatisticsCoordinator.start();
+    setAllTasksReady(NUM_SUBTASKS, dataStatisticsCoordinator, receivingTasks);
+  }
+
+  @Test
+  public void testThrowExceptionWhenNotStarted() {
+    String failureMessage =
+        "Call should fail when data statistics coordinator has not started yet.";
+
+    Assert.assertThrows(
+        failureMessage,
+        IllegalStateException.class,
+        () ->
+            dataStatisticsCoordinator.handleEventFromOperator(
+                0,
+                0,
+                DataStatisticsEvent.create(0, new MapDataStatistics(), statisticsSerializer)));
+    Assert.assertThrows(
+        failureMessage,
+        IllegalStateException.class,
+        () -> dataStatisticsCoordinator.executionAttemptFailed(0, 0, null));
+    Assert.assertThrows(
+        failureMessage,
+        IllegalStateException.class,
+        () -> dataStatisticsCoordinator.checkpointCoordinator(0, null));
+  }
+
+  @Test
+  public void testDataStatisticsEventHandling() throws Exception {
+    tasksReady();
+    RowType rowType = RowType.of(new VarCharType());
+    BinaryRowData binaryRowDataA =
```

Comment:
why do we need to convert `GenericRowData` to `BinaryRowData`? at least we should document the reason in comment.

This question applies to all usage of `BinaryRowData` in other test classes.